sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/CatalystTestUtils.scala

```diff
@@ -37,4 +40,32 @@ object DateTimeTestUtils {
       DateTimeUtils.resetThreadLocals()
     }
   }
+
+  /**
+   * Sets all SQL configurations specified in `pairs`, calls `f`, and then restore all SQL
+   * configurations.
+   */
+  protected def withSQLConf(pairs: (String, String)*)(f: => Unit): Unit = {
```

Comment:
Hi, @gatorsmile .
Then, what about removing `withSQLConf` from `SQLTestUtils` instead of having duplicated logic?
Since this is `trait`, I think we can do that in this PR, too.